src/authStrategies/RemoteAuth.js

```diff
@@ -0,0 +1,221 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const unzipper = require('unzipper');
+const archiver = require('archiver');
+const BaseAuthStrategy = require('./BaseAuthStrategy');
+
+/**
+ * Remote-based authentication
+ * @param {object} options - options
+ * @param {object} options.store - Remote database store instance
```

Comment:
Ok i see what you mean. Then stores would do something like _(using my MongoStore as example):_ 

```js
const { BaseStore } = require('whatsapp-web.js');
class MongoStore extends BaseStore {
...
```
Would it be ok to include `BaseStore` Class under `src/structures` ?

Reply:
> Ok i see what you mean. Then stores would do something like _(using my MongoStore as example):_
> 
> ```js
> const { BaseStore } = require('whatsapp-web.js');
> class MongoStore extends BaseStore {
> ...
> ```
> 
> Would it be ok to include `BaseStore` Class under `src/structures` ?

yeah kind of like that but since it is an interface you use the implement keyword, and all the required methods would fail to compile the app if they were missing. would also have comments to explain what is what so the people making the stores could make it in a better way

Reply:
Actually, we already have the `Store` interface defined on wwebjs `index.d.ts` as shown below:

```js
/** 
 * Remote store interface
 */
export interface Store {
    sessionExists: ({session: string}) => Promise<boolean> | boolean,
    delete: ({session: string}) => Promise<any> | any,
    save: ({session: string}) => Promise<any> | any,
    extract: ({session: string, path: string}) => Promise<any> | any,
}
```

Therefore, new `stores` can already implement the interface defined. I think this one can be marked as resolved, right? We can extend its documentation ofc, but code-wise i believe this is already done.